contrib_bots/lib/wikipedia.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import absolute_import
+import requests
+
+# See readme.md for instructions on running this code.
+
+class WikipediaHandler(object):
+    '''
+    This plugin facilitates searching Wikipedia for a
+    specific key term and returns the top article from the
+    search. It looks for messages starting with '@wikipedia'
+     or '@wiki'.
+
+    In this example, we write all Wikipedia searches into
+    the same stream that it was called from, but this code
+    could be adapted to write Wikipedia searches to some
+    kind of external issue tracker as well.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to directly search
+            Wikipedia for a specific key term and get the top
+            article that is returned from the search. Users
+            should preface searches with "@wikipedia" or
+            "@wiki".
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting Wikipedia
+        # searches for own Wikipedia searches!
+        if message['sender_full_name'] == 'wikipedia-bot':
+            return False
+        is_wikipedia = (original_content.startswith('@wiki') or
+                        original_content.startswith('@wikipedia'))
+
+        return is_wikipedia
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+
+        if original_content.startswith('@wikipedia'):
+            original_content = original_content[len('@wikipedia '):]
+        elif original_content.startswith('@wiki'):
+            original_content = original_content[len('@wiki '):]
+
+        query_wiki_link = 'https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch='
+        try:
+            data = requests.get(query_wiki_link + original_content + '&format=json')
+        except:
+            return 1
+
+        if data.status_code != 200:
+            return 1
```

Comment:
Just say `return` here, because the caller doesn't know about numeric return codes.  If you get errors here, you can log them with `print()` for now.